tools/ksvalidator.py

```diff
@@ -80,42 +81,62 @@ def main(argv):
     if not opts.ksfile:
         return (1, op.format_usage().split("\n"))
 
-    destdir = tempfile.mkdtemp(prefix="ksvalidator-tmp-")
-    try:
-        f = load_to_file(opts.ksfile, os.path.join(destdir, "ks.cfg"))
-    except KickstartError as e:
-        return (cleanup(destdir),
-                [_("Error reading %(filename)s:\n%(version)s") % {"filename": opts.ksfile, "version": e}])
-
-    try:
-        handler = makeVersion(opts.version)
-    except KickstartVersionError:
-        return (cleanup(destdir),
-                [_("The version %s is not supported by pykickstart") % opts.version])
-
-    ksparser = KickstartParser(handler, followIncludes=opts.followincludes,
-                               errorsAreFatal=opts.firsterror)
-
-    # turn kickstart parse warnings into errors
-    warnings.filterwarnings(action="error", category=KickstartParseWarning)
-
-    processedFile = None
-
-    try:
-        processedFile = preprocessKickstart(f)
-        ksparser.readKickstart(processedFile)
-        return (cleanup(destdir, processedFile, exitval=ksparser.errorsCount), [])
-    except KickstartDeprecationWarning as err:
-        return (cleanup(destdir, processedFile),
-                [_("File uses a deprecated option or command.\n%s") % err])
-    except KickstartParseError as err:
-        return (cleanup(destdir, processedFile), [str(err)])
-    except KickstartError:
-        return (cleanup(destdir, processedFile),
-                [_("General kickstart error in input file")])
-    except Exception as e:                                      # pylint: disable=broad-except
-        return (cleanup(destdir, processedFile),
-                [_("General error in input file:  %s") % e])
+    retval = 0
+    retmsg = []
+
+    # unpack any globs
+    ksfiles = []
+    for inksfile in opts.ksfile:
+        if is_url(inksfile):
+            ksfiles.append(inksfile)
+        else:
+            ksfiles.extend(glob.glob(inksfile))
+
+    # check if there are any files to check
+    if not ksfiles:
+        return (1, "No files match the patterns.")
+
+    # iterate over files to check them
+    with tempfile.TemporaryDirectory(prefix="ksvalidator-tmp-") as destdir:
+        for ksfile in ksfiles:
+            print(_("Checking kickstart file %(filename)s\n" % {"filename": ksfile}))
```

Comment:
I am not sure why pylint is complaining about this %, we have _() strings with % in them all over the place.

Reply:
Hm. pocketlint should be keeping that warning from showing up, maybe something odd with the GH workflow setup.

Reply:
Doesn't show with last changes, probably a fluke of some kind.